translator-api/src/main/java/org/xwiki/contrib/translator/TranslatorScriptService.java

```diff
@@ -156,4 +159,34 @@ public DocumentReference computeTranslationReference(DocumentReference originalD
         }
         return translator.computeTranslationReference(originalDocument, translationTitle, translationLocale);
     }
+
+    public List<GlossaryInfo> getGlossaries() throws TranslatorException
+    {
+        Translator translator = translatorManager.getTranslator();
+        if (this.authorizationManager.hasAccess(Right.PROGRAM)) {
```

Comment:
Why do we need the PROGRAM right here?

Reply:
Well, if we consider that all field into `GlossaryInfo` is not critical we can remove the check  right.